ipa-core/src/protocol/ipa_prf/malicious_security/prover.rs

```diff
@@ -123,15 +123,22 @@ impl<F: PrimeField, const λ: usize, const P: usize, const M: usize> ProofGenera
     ///
     /// Distributed Zero Knowledge Proofs algorithm drawn from
     /// `https://eprint.iacr.org/2023/909.pdf`
-    fn compute_proof<J, B>(uv_iterator: J, lagrange_table: &LagrangeTable<F, λ, M>) -> [F; P]
+    ///
+    /// returns proof and number of summands `m` which is important for the verification
```

Comment:
Generally, when you find yourself writing that a function does X *and* Y, it should be two separate functions.